File: labs/step-functions/aws/nodejs/src/fetch/handler.js
exports.handler = async (event) => {
  if (event.simulateFailure) {
    throw new Error("Falha simulada em FetchCEP");
  }

  const cep = event.cep;
  const res = await fetch(`https://viacep.com.br/ws/${cep}/json/`);
  const data = await res.json();

  if (data.erro) {
    throw new Error(`CEP não encontrado: ${cep}`);
  }

  return { ...event, viacep: data };
};

File: labs/step-functions/aws/python/src/fetch/handler.py
import json
import urllib.request


def lambda_handler(event, context):
    if event.get("simulateFailure"):
        raise RuntimeError("Falha simulada em FetchCEP")

    cep = event["cep"]
    url = f"https://viacep.com.br/ws/{cep}/json/"

    with urllib.request.urlopen(url, timeout=10) as resp:
        data = json.loads(resp.read().decode())

    if data.get("erro"):
        raise ValueError(f"CEP não encontrado: {cep}")

    return {**event, "viacep": data}
